src/index.ts

```diff
@@ -306,8 +313,19 @@ class Geckoboard {
     return new Dataset(schema);
   }
 
-  ping(): Promise<void> {
-    return Promise.resolve();
+  async ping(): Promise<void> {
+    const base64UserString = btoa(`${this.apiKey}:`);
+    const res = await fetch('https://api.geckoboard.com', {
+      headers: {
+        Authorization: `Basic ${base64UserString}`,
+      },
+    });
+    if (res.status !== 200) {
```

Comment:
I'm not sure I follow. If we make a request to `https://api.geckoboard.com` and it's successful it will have status 200 - does that not mean we should only accept 200, as any other status means something different has happened? Would we get e.g 202 and if so, why?

Reply:
I can't entirely remember what status codes the Datasets API returns, I was mostly just saying that what you're actually interested in is whether the status code is 2xx — which is exactly what `res.ok` would do.